pkg/authorization/registry/localresourceaccessreview/rest_test.go

```diff
@@ -0,0 +1,186 @@
+	authorizer    *testAuthorizer
+	reviewRequest *authorizationapi.LocalResourceAccessReview
+}
+
+type testAuthorizer struct {
+	users  util.StringSet
+	groups util.StringSet
+	err    string
+
+	actualAttributes authorizer.DefaultAuthorizationAttributes
+}
+
+func (a *testAuthorizer) Authorize(ctx kapi.Context, attributes authorizer.AuthorizationAttributes) (allowed bool, reason string, err error) {
+	// allow the initial check for "can I run this RAR at all"
+	if attributes.GetResource() == "localresourceaccessreviews" {
+		return true, "", nil
+	}
+
+	return false, "", errors.New("Unsupported")
+}
+func (a *testAuthorizer) GetAllowedSubjects(ctx kapi.Context, passedAttributes authorizer.AuthorizationAttributes) (util.StringSet, util.StringSet, error) {
+	attributes, ok := passedAttributes.(authorizer.DefaultAuthorizationAttributes)
+	if !ok {
+		return nil, nil, errors.New("unexpected type for test")
+	}
+
+	a.actualAttributes = attributes
+	if len(a.err) == 0 {
+		return a.users, a.groups, nil
+	}
+	return a.users, a.groups, errors.New(a.err)
+}
+
+func TestNoNamespace(t *testing.T) {
+	test := &resourceAccessTest{
+		authorizer: &testAuthorizer{
+			err: "namespace is required on this type: ",
+		},
+		reviewRequest: &authorizationapi.LocalResourceAccessReview{
+			Action: authorizationapi.AuthorizationAttributes{
+				Namespace: "",
+				Verb:      "get",
+				Resource:  "pods",
+			},
+		},
+	}
+
+	test.runTest(t)
+}
+
+func TestConflictingNamespace(t *testing.T) {
+	authorizer := &testAuthorizer{}
+	reviewRequest := &authorizationapi.LocalResourceAccessReview{
+		Action: authorizationapi.AuthorizationAttributes{
+			Namespace: "foo",
+			Verb:      "get",
+			Resource:  "pods",
+		},
+	}
+
+	storage := NewREST(resourceaccessreview.NewRegistry(resourceaccessreview.NewREST(authorizer)))
+	ctx := kapi.WithNamespace(kapi.NewContext(), "bar")
+	_, err := storage.Create(ctx, reviewRequest)
+	if err == nil {
+		t.Fatalf("unexpected non-error: %v", err)
+	}
+	if e, a := "namespace: invalid value 'foo', Details: namespace must be: bar", err.Error(); e != a {
+		t.Fatalf("expected %v, got %v", e, a)
+	}
+}
+
+func TestEmptyReturn(t *testing.T) {
+	test := &resourceAccessTest{
+		authorizer: &testAuthorizer{
+			users:  util.StringSet{},
+			groups: util.StringSet{},
+		},
+		reviewRequest: &authorizationapi.LocalResourceAccessReview{
+			Action: authorizationapi.AuthorizationAttributes{
+				Namespace: "unittest",
+				Verb:      "get",
+				Resource:  "pods",
+			},
+		},
+	}
+
+	test.runTest(t)
+}
+
+func TestNoErrors(t *testing.T) {
+	test := &resourceAccessTest{
+		authorizer: &testAuthorizer{
+			users:  util.NewStringSet("one", "two"),
+			groups: util.NewStringSet("three", "four"),
+		},
+		reviewRequest: &authorizationapi.LocalResourceAccessReview{
+			Action: authorizationapi.AuthorizationAttributes{
+				Namespace: "unittest",
+				Verb:      "delete",
+				Resource:  "deploymentConfig",
+			},
+		},
+	}
+
+	test.runTest(t)
```

Comment:
Every place you do this should just be a table test.  I don't think there is any benefit to splitting them this way (especially since your log messages are still going to be wrong).